components/ActiveStream/index.tsx

```diff
@@ -0,0 +1,495 @@
+          </Typography>
+        </Box>
+      </Box>
+      <Typography variant="body2" color="textSecondary">
+        100%
+      </Typography>
+    </Box>
+  );
+};
+
+function PaperComponent(props: PaperProps | undefined) {
+  const ref = useRef<HTMLDivElement>(null);
+  return (
+    <Draggable nodeRef={ref} cancel={'[class*="MuiDialogContent-root"]'}>
+      <Paper {...props} ref={ref} />
+    </Draggable>
+  );
+}
+
+const Transition = React.forwardRef(function Transition(
+  props: TransitionProps & {
+    children: React.ReactElement;
+  },
+  ref: React.Ref<unknown>,
+) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+function Modal({ open, onClose, children }: PropsWithChildren<{ open: boolean; onClose: () => void }>) {
+  const { breakpoints } = useTheme();
+  const isMobile = useMediaQuery(breakpoints.down('sm'));
+
+  return (
+    <Dialog
+      data-testid="vesting-vest-modal"
+      open={open}
+      onClose={onClose}
+      PaperComponent={PaperComponent}
+      PaperProps={{
+        sx: {
+          borderRadius: 2,
+          minWidth: '375px',
+        },
+      }}
+      TransitionComponent={isMobile ? Transition : undefined}
+      fullScreen={isMobile}
+      sx={isMobile ? { top: 'auto' } : { backdropFilter: 'blur(1.5px)' }}
+    >
+      <IconButton
+        aria-label="close"
+        onClick={onClose}
+        sx={{
+          position: 'absolute',
+          right: 4,
+          top: 8,
+          color: 'grey.500',
+        }}
+        data-testid="vesting-vest-modal-close"
+      >
+        <CloseIcon sx={{ fontSize: 19 }} />
+      </IconButton>
+      {children}
+    </Dialog>
+  );
+}
+
+type ActiveStreamProps = {
+  tokenId: number;
+  depositAmount: bigint;
+  withdrawnAmount: bigint;
+  startTime: number;
+  endTime: number;
+  cancellable: boolean;
+  refetch: () => void;
+};
+
+const ActiveStream: FC<ActiveStreamProps> = ({
+  tokenId,
+  depositAmount,
+  withdrawnAmount,
+  startTime,
+  endTime,
+  cancellable,
+  refetch,
+}) => {
+  const { t } = useTranslation();
+  const { impersonateActive, chain: displayNetwork, opts } = useWeb3();
+  const { chain } = useNetwork();
+  const { spacing } = useTheme();
+  const { switchNetwork, isLoading: switchIsLoading } = useSwitchNetwork();
+  const { data: reserve, isLoading: reserveIsLoading } = useEscrowedEXAReserves(BigInt(tokenId));
+  const { data: withdrawable, isLoading: withdrawableIsLoading } = useSablierV2LockupLinearWithdrawableAmountOf(
+    BigInt(tokenId),
+  );
+  const escrowedEXA = useEscrowedEXA();
+  const [loading, setLoading] = useState(false);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [modalContent, setModalContent] = useState<'nft' | 'cancel'>('nft');
+  const { breakpoints } = useTheme();
+  const isMobile = useMediaQuery(breakpoints.down('sm'));
+  const { data: nft } = useSablierV2NftDescriptorTokenUri(BigInt(tokenId));
+
+  const WithdrawAndCancel = () => {
```

Comment:
please, dont declare a component inside another component, as this will be re-created on each render, declare outside and pass props